src/main/java/nl/tudelft/ewi/devhub/server/web/resources/repository/AbstractProjectResource.java

```diff
@@ -600,29 +604,56 @@ public Response showSettings(@Context HttpServletRequest request) throws IOExcep
 	@POST
 	@Path("/branches/delete")
 	@Transactional
-	public Response deleteBehindBranch(@Context HttpServletRequest request,
-									   @FormParam("branchDeleteName") String branchDeleteName,
-									   @QueryParam("fatal") String fatal)
+	public Response deleteBranch(@Context HttpServletRequest request,
+                                 @FormParam("branchDeleteName") String branchDeleteName,
+                                 @FormParam("branchDeleteNameConfirmation")
+                                             String branchDeleteNameConfirmation,
+                                 @QueryParam("fatal") String fatal)
 			throws IOException, ApiError {
 
 		RepositoryEntity repositoryEntity = getRepositoryEntity();
 		RepositoryApi repositoryApi = repositoriesApi.getRepository(repositoryEntity.getRepositoryName());
 
         List<Locale> locales = Collections.list(request.getLocales());
 
-        Map<String, Object> parameters = getBaseParameters();
+        Map<String, Object> parameters;
 
         if (branchDeleteName != null) {
             BranchApi branchApi = repositoryApi.getBranch(branchDeleteName);
 			BranchModel branchModel = branchApi.get();
 
 			if (!branchModel.isAhead()) {
+                // Branch is not ahead and can be safely removed
 				branchApi.deleteBranch();
-				parameters = getBranchOverviewParameters("master", 1, true);
+				parameters = getBranchOverviewParameters("master", 1, "success");
 			} else {
-				parameters = getBranchOverviewParameters("master", 1, false);
+                // Branch is ahead
+                if (branchDeleteNameConfirmation != null) {
+					String branchSimpleName = branchDeleteName.split("refs/heads/")[1];
+                    if (branchSimpleName.equals(branchDeleteNameConfirmation)
+							&& !branchDeleteName.equals("refs/heads/master")) {
+                        // Confirmation is correct and branch is not master, delete ahead branch.
+                        branchApi = repositoryApi.getBranch(branchDeleteName);
+                        branchModel = branchApi.get();
+                        branchApi.deleteBranch();
+                        parameters = getBranchOverviewParameters("master", 1, "success");
+                        parameters.put("deletedBranchSimpleName", branchModel.getSimpleName());
```

Comment:
As `deletedBranchName` and `aheadBranchName` are comming from the same branch Model just use `branchModel` and use an other variable as flag.